ratis-server/src/main/java/org/apache/ratis/server/raftlog/RaftLogBase.java

```diff
@@ -274,7 +274,7 @@ public final void open(long lastIndexInSnapshot, Consumer<LogEntryProto> consume
         e -> commitIndex.updateToMax(e.getMetadataEntry().getCommitIndex(), infoIndexChange));
     state.open();
 
-    final long startIndex = getStartIndex();
+    final long startIndex = Math.max(getStartIndex(), lastIndexInSnapshot + 1);
     if (startIndex > LEAST_VALID_LOG_INDEX) {
       purgeIndex.updateIncreasingly(startIndex - 1, infoIndexChange);
```

Comment:
I am thinking that `purgeIndex` should not depend on `lastIndexInSnapshot`.  How about changing to `purgeIndex.updateToMax` instead?
```java
      purgeIndex.updateToMax(startIndex - 1, infoIndexChange);
```

Reply:
Great idea！Has fixed